src/Http/Routing/src/RouteOptions.cs

```diff
@@ -152,7 +159,7 @@ public void SetParameterPolicy(string token, [DynamicallyAccessedMembers(Dynamic
         _constraintTypeMap[token] = type;
     }
 
-    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(Dictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
+    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(IDictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
```

Comment:
undo this change:

```suggestion
    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(Dictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
```

Reply:
I either do this, or I cast from ```IDictionary<string, Type>``` to ```Dictionary<string, Type>```. In the initial setup of the constraint map there is a ```Dictionary<string, Type>``` but it is persisted in an ```IDictionary<string, Type>``` field which I then need to go an add the regex and alpha constraints to at some point in the future.

Was there a reason for your concern around changing this private method signature?

Reply:
Ah I see, it's exposed publicly. It's fine then.